### PR TITLE
Correction for documentation of FunctionSpace

### PR DESCRIPTION
**Pull request by juan42**
dof_count and node_count do return the number of _local_ degrees of freedom

### CODE REVIEW

File: firedrake/functionspaceimpl.py [hunk 1]
```diff
@@ -391,15 +391,15 @@ def __mul__(self, other):
 
     @property
     def node_count(self):
-        """The number of global nodes in the function space.  If the
+        """The number of local nodes in the function space.  If the
```
Comment (dham) [inline firedrake/functionspaceimpl.py]: I think there might be a degree of confusion about nomenclature here. I think "local" means "per element".

Comment (dham) [inline firedrake/functionspaceimpl.py]: How about "The number of nodes of this function space on this process."

Comment (miklos1) [inline firedrake/functionspaceimpl.py]: How about "The number of nodes of this function space _visible_ on this process."

**Comment by wence-** [comment]
Maybe add a parenthesised ("includes halo nodes") ?

**Comment by wence-** [comment]
Please fix the trailing whitespace.

**Pull request merged**